daemons/fenced/fenced_commands.c

```diff
@@ -2081,42 +2081,43 @@ localhost_is_eligible(const stonith_device_t *device, const char *action,
 }
 
 static void
-can_fence_host_with_device(stonith_device_t * dev, struct device_search_s *search)
+can_fence_host_with_device(stonith_device_t *dev,
+                           struct device_search_s *search)
 {
     gboolean can = FALSE;
     const char *check_type = NULL;
-    const char *host = search->host;
+    const char *target = NULL;
     const char *alias = NULL;
+    const char *action = (search == NULL)? NULL : search->action;
 
-    CRM_LOG_ASSERT(dev != NULL);
+    CRM_CHECK((dev != NULL) && (action != NULL), goto search_report_results);
 
-    if (dev == NULL) {
-        goto search_report_results;
-    } else if (host == NULL) {
+    target = search->host;
+    if (target == NULL) {
         can = TRUE;
         goto search_report_results;
     }
 
     /* Short-circuit query if this host is not allowed to perform the action */
-    if (pcmk__str_eq(search->action, "reboot", pcmk__str_casei)) {
+    if (pcmk__str_eq(action, "reboot", pcmk__str_casei)) {
         /* A "reboot" *might* get remapped to "off" then "on", so short-circuit
          * only if all three are disallowed. If only one or two are disallowed,
          * we'll report that with the results. We never allow suicide for
          * remapped "on" operations because the host is off at that point.
          */
-        if (!localhost_is_eligible(dev, "reboot", host, search->allow_suicide)
-            && !localhost_is_eligible(dev, "off", host, search->allow_suicide)
-            && !localhost_is_eligible(dev, "on", host, FALSE)) {
+        if (!localhost_is_eligible(dev, "reboot", target, search->allow_suicide)
+            && !localhost_is_eligible(dev, "off", target, search->allow_suicide)
+            && !localhost_is_eligible(dev, "on", target, FALSE)) {
             goto search_report_results;
```

Comment:
when remapping we might remap "reboot" to a sole "off" but shouldn't unavailability of both "reboot" & "off" already be a hard criterion that it won't fit?

Reply:
I'm not 100% sure, but I believe it's possible for different devices (or at least nodes) to be chosen for the "off" and "on" portions, so it's still worth telling the initiator if this node can only do one of them

Reply:
Hadn't checked for the actual flow of things and can't recall from memory.